### PR TITLE
Add task filter

### LINKED ISSUE #95: Implement Task Filter

### PR DESCRIPTION
**Pull request by T-Fang**
Fixes #95

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -520,52 +531,93 @@ Use case ends.
...
+  Use case ends.
+
+**Use case: UC13 - View Tasks of a Person**
 
 **Actor:** User
 
 **MSS:**
 
-1. User requests to view all tasks assigned to a teammate.
+1. User requests to view all tasks assigned to a teammate by filtering tasks (UC12).
```
Comment (GeNiaaz) [inline docs/DeveloperGuide.md:599]: good one, makes it more specific 👍

File: docs/UserGuide.md [hunk 2]
```diff
@@ -203,6 +207,22 @@ Format: `assign INDEX NAME`
 
 Example: `assign 1 Niaaz` Assigns the task currently with index 1 to Niaaz.
 
+#### Filter tasks by assignee/deadline/task name `filter `
+
+Filter tasks by assignee, deadline or task name. 
+
+Format: ``filter (ta/ASSIGNEE NAME)||(td/DEADLINE)||(tn/TASK NAME)``
+
+- `filter (ta/ASSIGNEE NAME)` retrieves all tasks that has assignee named `ASSIGNEE NAME`
```
Comment (GeNiaaz) [inline docs/UserGuide.md]: small grammatical error here: `that have assignee named...`

File: docs/UserGuide.md [hunk 3]
```diff
@@ -203,6 +207,22 @@ Format: `assign INDEX NAME`
...
+- `filter (td/DEADLINE)` retrieves all tasks whose deadline is`DEADLINE`
+- `filter (tn/TASK NAME)` retrieves all tasks whose task name contains `TASK NAME`
+
+Example: 
+
+1. `filter ta/Alice` Displays filtered list of tasks that has assignee named Alice.
```
Comment (GeNiaaz) [inline docs/UserGuide.md]: same grammatical error here: `that have assignee named...`

File: docs/UserGuide.md [hunk 4]
```diff
@@ -231,7 +251,7 @@ Format: `updatetm NAME [n/NAME] [p/PHONE_NUMBER] [e/EMAIL]`
 
 Example: `updatetm Lucas p/12345678` Resets the contact number of `Lucas` to `12345678`
 
-#### Update participation details (tasks and project-specific information) `updatept participation `
+#### Update participation details (tasks and project-specific information) `update participation `
```
Comment (GeNiaaz) [inline docs/UserGuide.md:254]: good spot! 💯

File: src/main/java/seedu/address/logic/commands/project/FilterCommand.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+import seedu.address.model.project.Project;
+import seedu.address.model.task.Task;
+
+/**
+ * Filters tasks by assignee's name, task name or deadline.
+ */
+public class FilterCommand extends Command {
+    public static final String COMMAND_WORD = "filter";
+
+    public static final String MESSAGE_NOT_IMPLEMENTED_YET = "Filter command not implemented yet";
```
Comment (GeNiaaz) [inline src/main/java/seedu/address/logic/commands/project/FilterCommand.java]: You haven't implemented it yet?

**Review by GeNiaaz — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/logic/commands/project/FilterCommand.java (hunk 5 above)
Comment (T-Fang) [reply src/main/java/seedu/address/logic/commands/project/FilterCommand.java]: I forgot to delete it, I will delete it later

**Review by GeNiaaz — APPROVED** [review]
Looks good! 👍

**Pull request merged**